gaphor/SysML/drop.py

```diff
@@ -11,7 +13,23 @@ def drop_proxy_port(element: ProxyPort, diagram, x, y):
     if not item_class:
         return None
 
-    head_item = diagram_has_presentation(diagram, element.encapsulatedClassifier)
+    def property_item_which_is_typed_by_block_owning_the_port(item):
+        return (
+            isinstance(item, PropertyItem)
+            and isinstance(item.subject.type, Block)
+            and element in item.subject.type.ownedPort
+        )
+
+    head_item = next(
+        (
+            item
+            for item in diagram.select(
+                property_item_which_is_typed_by_block_owning_the_port
```

Comment:
I would inline the code.

```suggestion
                lambda e: isinstance(e, PropertyItem)
                and isinstance(e.subject.type, Block)
                and element in e.subject.type.ownedPort
```